src/Utils.ps1

```diff
@@ -289,10 +290,24 @@ function Get-PromptPath {
     $pathInfo = $ExecutionContext.SessionState.Path.CurrentLocation
     $currentPath = if ($pathInfo.Drive) { $pathInfo.Path } else { $pathInfo.ProviderPath }
 
+    # Look up the git root
+    if ($abbrevGitDir) {
+        $gitPath = $(Get-GitDirectory)
+        # Up one level from `.git`
+        if ($gitPath) { $gitPath = $(Split-Path $gitPath -Parent) }
+    }
+
     $stringComparison = Get-PathStringComparison
 
```

Comment:
Let's simplify the following logic a bit by returning early if $currentPath is null or empty e.g.:
```
if (!$currentPath) { return $currentPath }
```
This is semantically equivalent and you can get rid of the `$currentPath` term in both the following `if` and `elseif`.